packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -267,6 +272,13 @@ export class AccountTreeController extends BaseController<
     this.update((state) => {
       state.accountTree.wallets = wallets;
 
+      // Apply group metadata within the state update
+      for (const wallet of Object.values(wallets)) {
```

Comment:
As discussed internally, we should be using `state` here, otherwise the parameters we pass down to `#applyAccountGroupMetadata` will reference the `wallets` ref, which won't mutate the `state.accountTree` names.

Though, the persisted names are good 👍

```suggestion
      for (const wallet of Object.values(state.accountTree.wallets)) {
```

Reply:
Actually.. now that we pass the `state` to `#applyAccountGroupMetadata` we could maybe re-use it that directly?

So the caller is not expected to pass a reference to the `immer` draft directly. That sounds better to me, WDYT?